Description: unmarked assembly causes the stack to be marked as executable.
 This upstream patch fixes the assembly behavior.
Ubuntu: https://launchpad.net/bug/409864
Upstream: https://bugzilla.mozilla.org/show_bug.cgi?id=486537

Index: nss-3.12.3.1/mozilla/security/nss/lib/freebl/Makefile
===================================================================
--- nss-3.12.3.1.orig/mozilla/security/nss/lib/freebl/Makefile	2009-08-24 15:01:51.000000000 -0700
+++ nss-3.12.3.1/mozilla/security/nss/lib/freebl/Makefile	2009-08-24 15:03:01.000000000 -0700
@@ -145,7 +145,7 @@
 ifeq ($(OS_TARGET),Linux)
 ifeq ($(CPU_ARCH),x86_64)
     ASFILES  = arcfour-amd64-gas.s mpi_amd64_gas.s
-    ASFLAGS += -march=opteron -m64 -fPIC
+    ASFLAGS += -march=opteron -m64 -fPIC -Wa,--noexecstack
     DEFINES += -DNSS_BEVAND_ARCFOUR -DMPI_AMD64 -DMP_ASSEMBLY_MULTIPLY
     DEFINES += -DNSS_USE_COMBA
     DEFINES += -DMP_CHAR_STORE_SLOW -DMP_IS_LITTLE_ENDIAN
